gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,217 @@
+from gnomad_qc.v3.resources.sample_qc import (
+    hard_filtered_samples,
+    pca_related_samples_to_drop,
+)
+from gnomad.resources.grch38.reference_data import lcr_intervals
+from gnomad.utils.annotations import get_adj_expr
+from gnomad.utils.file_utils import file_exists
+from gnomad.utils.sparse_mt import densify_sites
+from gnomad.sample_qc.pipeline import get_qc_mt
+from gnomad_qc.v3.resources import release_sites
+from gnomad_qc.v3.resources.basics import get_gnomad_v3_mt
+from gnomad_qc.v3.resources.annotations import get_info, last_END_position
+from gnomad_qc.v3.resources.sample_qc import get_sample_qc_root
+
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("subpop_analysis")
+logger.setLevel(logging.INFO)
+
+
+def compute_subpop_qc_mt(
+    mt: hl.MatrixTable, min_popmax_af: float = 0.001,
+) -> hl.MatrixTable:
+    """
+    Generate the subpop QC MT to be used for the subpop analyses.
+    
+    Filter MT to bi-allelic SNVs at sites with popmax allele frequency greater than `min_popmax_af`, densify, and write MT.
+
+    :param mt: Raw MatrixTable to use for the subpop analysis
+    :param min_popmax_af: Minimum population max variant allele frequency to retain variant for the subpop QC MatrixTable
+    :return: MatrixTable filtered to variants for subpop analysis
+    """
+    release_ht = hl.read_table(release_sites().path)
+
+    # Filter to biallelic SNVs not in low-confidence regions and with a popmax above min_popmax_af
+    qc_sites = release_ht.filter(
+        (release_ht.popmax.AF > min_popmax_af)
+        & ~release_ht.was_split
+        & hl.is_snp(release_ht.alleles[0], release_ht.alleles[1])
+        & hl.is_missing(lcr_intervals.ht()[release_ht.locus])
+    ).key_by("locus")
+
+    # Densify the MT and include only sites defined in the qc_sites ht
+    mt = mt.select_entries(
+        "END", GT=mt.LGT, adj=get_adj_expr(mt.LGT, mt.GQ, mt.DP, mt.LAD)
+    )
+    mt = densify_sites(mt, qc_sites, hl.read_table(last_END_position.path))
+
+    return mt
+
+
+def run_pop_pca(
+    mt: hl.MatrixTable,
+    pop: str,
+    min_af: float = 0.001,
+    min_inbreeding_coeff_threshold: float = -0.25,
+    remove_hard_filtered_samples: bool = True,
+    release=False,
+    high_quality=False,
+    outliers=None,
+    mt_read_if_exists=True,
+    ht_read_if_exists=True,
+    outlier_string="",
+) -> hl.MatrixTable:
+    """
+    Generate the QC MT per specified population and generate PCA info.
+
+    :param mt: The QC MT output by the 'compute_subpop_qc_mt' function
+    :param min_af: Minimum variant allele frequency to retain variant in QC matrix table
+    :param min_inbreeding_coeff_threshold: Minimum site inbreeding coefficient to keep
+    :return: MatrixTable filtered to specific population to use for subpop analysis
+    """
+    # Adjust names for different project subpops that should actually be the same
```

Comment:
I'm thinking we don't actually need to do this anymore, Mike recommended using `subpop_description` instead: https://atgu.slack.com/archives/CRA2TKTV0/p1627330748010900?thread_ts=1626888229.003000&cid=CRA2TKTV0  so let's just remove below too. If we want to keep, we might want to just change this in the meta HT